File: src/main/java/de/mroedig/domain/dao/impl/BlubbsDaoImpl.java
package de.mroedig.domain.dao.impl;

import de.mroedig.domain.dao.BlubbsDao;
import de.mroedig.domain.entity.Blubbs;
import org.springframework.stereotype.Repository;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;

@Repository
public class BlubbsDaoImpl implements BlubbsDao {

    @PersistenceContext
    private EntityManager em;

    @Override
    public void persist(Blubbs b) {
        em.persist(b);
    }
}

File: src/main/kotlin/de/mroedig/domain/dao/impl/BlubbsDaoImpl.kt
package de.mroedig.domain.dao.impl

import de.mroedig.domain.dao.BlubbsDao
import de.mroedig.domain.entity.Blubbs
import org.springframework.stereotype.Repository
import javax.persistence.EntityManager
import javax.persistence.PersistenceContext

@Repository
class BlubbsDaoImpl : BlubbsDao {
    @PersistenceContext
    private val em: EntityManager? = null

    override fun persist(b: Blubbs?) {
        em!!.persist(b)
    }
}